resources/volume.rb

```diff
@@ -27,16 +27,30 @@
   # @return [OneviewSDK::Volume] Loaded Volume resource
   def load_resource_with_associated_resources
     item = load_resource
-    item = load_storage_system(item)
+    load_storage_system(item)
 
     # item.set_storage_pool(OneviewSDK::StoragePool.new(item.client, name: storage_pool)) if storage_pool
-    # HACK Ruby SDK 1.0.0 issue workaround
-    sp = OneviewSDK::StoragePool.new(item.client, name: storage_pool)
-    raise "Storage Pool '#{sp['name']}' not found" unless sp.retrieve!
-    item['provisioningParameters']['storagePoolUri'] = sp['uri']
+    # Workaround for issue in oneview-sdk:
+    if storage_pool
+      sp = OneviewSDK::StoragePool.find_by(item.client, name: storage_pool).first
+      raise "Storage Pool '#{sp['name']}' not found" unless sp
+      item['storagePoolUri'] = sp['uri']
+    end
 
     item.set_snapshot_pool(OneviewSDK::StoragePool.new(item.client, name: snapshot_pool)) if snapshot_pool
-    item.set_volume_template(OneviewSDK::VolumeTemplate.new(item.client, name: volume_template)) if volume_template
+    item.set_storage_volume_template(OneviewSDK::VolumeTemplate.new(item.client, name: volume_template)) if volume_template
+
+    # Convert capacity integers to strings
+    item['provisionedCapacity'] = item['provisionedCapacity'].to_s if item['provisionedCapacity']
+    item['allocatedCapacity'] = item['allocatedCapacity'].to_s if item['allocatedCapacity']
```

Comment:
You shouldn't have to set it, but if a user would happen to set it as an integer instead of a string, it would cause the comparison to report that they're different. This just converts it if they set it. If not, it does nothing. (Makes it a little more fool-proof :smile: )